### ISSUE TITLE
False positive CS8620 with extension+params

### ISSUE THREAD

**Issue by gurustron** [dotnet/roslyn#81396]
**Version Used**:  10.0.100

**Steps to Reproduce**:

```
// Source - https://stackoverflow.com/a/79826426
// Posted by Jon Skeet

// no warning for old style extension
obj.OldMethod("x", "y");
//  warning CS8620: Argument of type 'string' cannot be used for parameter 'scopes' of type 'string[]' in 'void extension(object).NewMethod(params string[] scopes)' due to differences in the nullability of reference types.
obj.NewMethod("x", "y");
// No warning
obj.NewMethod(new[] { "x", "y" });

public static class Extensions
{
    public static void OldMethod(this object x, params string[] scopes) { }

    extension(object obj)
    {
        public void NewMethod(params string[] scopes) {}
    }
}
```

A minimal repro, with source-code provided, is ideal.  Most compiler/language issues can be distilled into a snippet that can be pasted into [.NET Lab](https://lab.razor.fyi/#csharp).

**Diagnostic Id**: CS8620: Argument of type 'string' cannot be used for parameter 'scopes' of type 'string[]' in 'void extension(object).NewMethod(params string[] scopes)' due to differences in the nullability of reference types.

**Expected Behavior**: no warning

**Actual Behavior**: warning

**Comment by gurustron** [comment]
Duplicate https://github.com/dotnet/roslyn/issues/80024
